### PR TITLE
#54: Refactoring of TextIs, TextHasString and StartsWith.

### PR DESCRIPTION
**Pull request by dgroup**
#54

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#56](https://github.com/llorllale/cactoos-matchers/issues/56) is now in scope, role is `REV`

**Comment by llorllale** [comment]
@dgroup the task is to "refactor existing Text matchers" - not provide a new one (`StartsWith`)

**Comment by dgroup** [comment]
@llorllale, could please check the review [details](https://github.com/llorllale/cactoos-matchers/pull/56/files#diff-ad6ecb2d6645173d03b5aa9b61210022)? 
I think my commit message wasn't clear enough.

**Comment by dgroup** [comment]
@llorllale, could you please review the PR ?

File: src/test/java/org/llorllale/cactoos/matchers/TextHasStringTest.java [hunk 1]
```diff
@@ -57,8 +57,7 @@ public void hasClearDescriptionForFailedTest() throws Exception {
         final TextHasString matcher = new TextHasString(
             "ed076287532e86365e841e92bfc50d8c6"
         );
-        matcher.matchesSafely(hex);
-        matcher.describeMismatchSafely(hex, description);
+        matcher.matchesSafely(hex, description);
```
Comment (llorllale) [inline src/test/java/org/llorllale/cactoos/matchers/TextHasStringTest.java]: @dgroup why this change?

Comment (dgroup) [reply src/test/java/org/llorllale/cactoos/matchers/TextHasStringTest.java]: Because now the TestHasString is extending TypeSafeDiagnosticMatcher instead of TypeSafeMatcher.

**Comment by dgroup** [comment]
@llorllale, is there any other review comments?

**Comment by llorllale** [comment]
@rultor merge

**Comment by llorllale** [comment]
@dgroup shoot: we forgot to remove the puzzle

**Comment by 0crat** [comment]
The job [#56](https://github.com/llorllale/cactoos-matchers/issues/56) is now out of scope
